File: src/main/java/com/jrmplugin/core/zip/ZipArchiveTask.java
package com.jrmplugin.core.zip;

import com.intellij.openapi.diagnostic.Logger;
import net.lingala.zip4j.ZipFile;
import net.lingala.zip4j.model.ZipParameters;
import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.io.IOException;
import java.util.UUID;

import static org.assertj.core.util.Files.newTemporaryFolder;

public class ZipArchiveTask {

    private static final Logger LOG = Logger.getInstance(ZipArchiveTask.class);

    public File zip(String path) {
        try {
            ZipFile zipFile = new ZipFile(createTemporaryZipFile());
            zipFile.addFolder(new File(path), prepareZipParameters());
            return zipFile.getFile();
        } catch (IOException ex) {
            LOG.error("Can't zip file: " + path);
            return null;
        }
    }

    @NotNull
    private ZipParameters prepareZipParameters() {
        ZipParameters zipParameters = new ZipParameters();
        zipParameters.setExcludeFileHandler(
                file -> file.getName().contains("target")
        );
        return zipParameters;
    }

    @NotNull
    private String createTemporaryZipFile() {
        return newTemporaryFolder() + File.separator + UUID.randomUUID().toString() + ".zip";
    }

}

File: src/main/java/com/jrmplugin/core/zip/ZipArchiveTask.kt
package com.jrmplugin.core.zip

import com.intellij.openapi.diagnostic.Logger
import net.lingala.zip4j.ZipFile
import net.lingala.zip4j.model.ExcludeFileHandler
import net.lingala.zip4j.model.ZipParameters
import java.io.File
import java.io.IOException
import java.nio.file.Files
import java.util.*

class ZipArchiveTask {
    fun zip(path: String): File? {
        return try {
            val zipFile = ZipFile(createTemporaryZipFile())
            zipFile.addFolder(File(path), prepareZipParameters())
            zipFile.file
        } catch (ex: IOException) {
            LOG.error("Can't zip file: $path")
            null
        }
    }

    private fun prepareZipParameters(): ZipParameters {
        val zipParameters = ZipParameters()
        zipParameters.excludeFileHandler = ExcludeFileHandler { file: File -> file.name.contains("target") }
        return zipParameters
    }

    private fun createTemporaryZipFile(): String {
        return try {
            (Files.createTempDirectory("tmp")
                    .toAbsolutePath().toString()
                    + File.separator + UUID.randomUUID().toString() + ".zip")
        } catch (e: IOException) {
            throw IllegalStateException()
        }
    }

    companion object {
        private val LOG = Logger.getInstance(ZipArchiveTask::class.java)
    }
}